R/labels.R

```diff
@@ -87,28 +102,141 @@ uses_labels <- function(p, ...) {
     )
   }
 
+  default_labels <- purrr::map_lgl(args, inherits, ".default_label")
+
+  args[default_labels] <- purrr::map(
+    names(args)[default_labels], ~ unlist(default_label(p, .))
+  )
+
   if (!all(is_scalar_string_or_null(args))) {
     stop(
       "All inputs to `...` must be character vectors of length 1 or `NULL`.",
       call. = FALSE
     )
   }
 
-  if (is.null(names(args)) || any(names(args) == "")) {
-    stop("All inputs to `...` must be named.", call. = FALSE)
+  if (is.null(names(args))) {
+    names(args) <- rep("",    length(args))
+    named       <- rep(FALSE, length(args))
+  } else {
+    named <- names(args) != ""
   }
 
-  labels <- get_labels(p, names(args))
+  result         <- logical(length(args))
+  result[!named] <- check_labels_set(p, args[!named])
```

Comment:
`args` make sense in this function as an object name since we've collected named and unnamed things from `...`, but once they're passed to `check_labels_set()` or `check_labels_match()` I think we need a better name for the object in those functions.

```r
# instead of
check_labels_set <- function(p, args) { ... }

# something like
check_labels_set <- function(p, labels) { ... }

# and/or
check_labels_match <- function(p, label_values) { ... }
```

Reply:
Resolved in 3b9a20f